test/e2e-tests.sh

```diff
@@ -85,6 +85,38 @@ $(dirname $0)/upload-test-images.sh e2e || fail_test "Error uploading test image
 # Setup resources common to all eventing tests
 setup_events_test_resources|| fail_test "Error setting up test resources"
 
-go_test_e2e ./test/e2e || fail_test
+go_test_e2e ./test/e2e
+exit_result=$?
+if [ ${exit_result} -ne 0 ]; then
+  # Collecting logs from all knative's eventing pods
+  echo "====== Collecting logs for namespace: knative-eventing ========"
+  for pod in $(kubectl get pod -n knative-eventing | grep Running | awk '{print $1}' ); do
+    echo "========== Collecting logs for pod: "${pod}" ================="
+    for container in $(kubectl get pod "${pod}" -n knative-eventing -ojsonpath='{.spec.containers[*].name}'); do
+      echo "----------------------------------------------------------"
+      echo "Container: "${container}
+      kubectl logs -n knative-eventing "${pod}" -c "${container}" || true
+      echo "----------------------------------------------------------"
+      kubectl logs -p -n knative-eventing "${pod}" -c "${container}" || true
+      echo "----------------------------------------------------------"
+    done
+    echo "============================================================"
+  done
+  # Collecting logs from the test namespace pods.
```

Comment:
Why is this split? Why not a for loop over the two namespaces?

Reply:
because the pods do very different things was my thinking

Reply:
The pods may do different things, but we are collecting logs in exactly the same way.

Reply:
I was not that excited about bash and making functions so copy pasta

Reply:
You don't need a function, just an outer for-loop for the namespace.

```shell
if [ ${exit_result} -ne 0 ]; then
  # Collecting logs from all relevant pods.
  for namespace in "knative-eventing" "e2etestfn3"; do
    for pod in $(kubectl get pod -n $namespace | grep Running | awk '{print $1}' ); do
      ... 
    done
  done
  fail_test
fi
```

Although a function would probably be nicer.